include/utilities/FileStreamHandler.hpp

```diff
@@ -14,7 +14,13 @@ namespace utils
                 stream->open(path, std::ios::trunc);
                 output_stream = stream;
             }
-            virtual ~FileStreamHandler(){}
+
+            virtual ~FileStreamHandler() override {
+                auto tmp = std::static_pointer_cast<std::ofstream>(output_stream);
+                if(tmp){
+                    tmp->close();
```

Comment:
I'm not remembering exactly why I did this (that's what I get for leaving it sitting in my working tree for so long 😵‍💫).  I think you are correct, and this may be an artifact of some early attempts to ensure the files were were closed but due to the shared references of formulations they were not and it took me a while to sort that out haha.  I'll take a closer look and remove this if it isn't being useful.